bl/interfaces/tournament-repo.go

```diff
@@ -0,0 +1,215 @@
+	if err != nil {
+		return nil, kerror.Newf(kerror.SQLPrepareStatementError, "prepare stmt %v: %v", query, err)
+	}
+	defer debugutil.Close(stmt)
+
+	if err := stmt.QueryRow(id).Scan(&tournament.ID, &tournament.Name, &tournament.Deposit, &tournament.Prize, &tournament.Winner, &tournament.Status); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, kerror.Newf(kerror.TournamentDoesntExists, "tournament with id(%v) isn't exists: %v", id, err)
+		}
+
+		return nil, kerror.Newf(kerror.SQLScanError, "scan query: %v", err)
+	}
+
+	users, err := tr.selectUserIDsOfTournament(store, id)
+	if err != nil {
+		return nil, kerror.Errorf(err, "get users of tournament")
+	}
+	tournament.Users = users
+
+	return tournament, nil
+
+}
+
+func (tr *TournamentRepository) selectUserIDsOfTournament(store tx.DBTX, tournamentID uuid.UUID) ([]models.User, error) {
+	const query = `
+		SELECT * FROM UsersOfTournaments WHERE tournamentID = $1;
+	`
+	users := []models.User{}
+
+	stmt, err := store.Prepare(query)
+	if err != nil {
+		return nil, kerror.Newf(kerror.SQLPrepareStatementError, "prepare query: %v", err)
+	}
+	defer debugutil.Close(stmt)
+
+	rows, err := stmt.Query(tournamentID)
+	if err != nil {
+		return nil, kerror.Newf(kerror.SQLConstraintError, "query request: %v", err)
+	}
+	defer debugutil.Close(rows)
+
+	for rows.Next() {
+		var user models.User
+
+		if err := rows.Scan(&user.ID, &user.Name, &user.Balance); err != nil {
+			return nil, kerror.Newf(kerror.SQLScanError, "scan user ids of tournament(%v): %v", tournamentID, err)
+		}
+
+		users = append(users, user)
+	}
+
+	return users, nil
+}
+
+func (tr *TournamentRepository) SelectRandomUserOfTournament(store tx.DBTX, tournamentID uuid.UUID) (*models.User, error) {
+	const query = `
+		WITH random_id AS (
+			SELECT user FROM UsersOfTournaments WHERE tournament = $1
+				OFFSET random() * COUNT(*) LIMIT 1
+		)
+		SELECT * FROM Users WHERER id = (SELECT id FROM random_id);
+	`
+	var user models.User
+
+	stmt, err := store.Prepare(query)
+	if err != nil {
+		return nil, kerror.Newf(kerror.SQLPrepareStatementError, "prepare stmt: %v", err)
+	}
+	defer debugutil.Close(stmt)
+
+	if err := stmt.QueryRow(tournamentID).Scan(&user.ID, &user.Name, &user.Balance); err != nil {
+		return nil, kerror.Newf(kerror.SQLScanError, "scan user from db: %v", err)
+	}
+
+	return &user, nil
+}
+
+func (tr *TournamentRepository) InsertUserToTournament(store tx.DBTX, tournamentID, userID uuid.UUID) error {
+	const query = `
+		INSERT INTO UsersOfTournaments(tournament, user) VALUES ($1, $2); 
+	`
+
+	stmt, err := store.Prepare(query)
+	if err != nil {
+		return kerror.Newf(kerror.SQLPrepareStatementError, "prepare query: %v", err)
+	}
+	defer debugutil.Close(stmt)
+
+	if _, err := stmt.Exec(tournamentID, userID); err != nil {
+		return kerror.Newf(kerror.SQLExecutionError, "exec stmt: %v", err)
+	}
+
+	return nil
+}
+
+func (tr *TournamentRepository) AddToPrize(store tx.DBTX, ID uuid.UUID, addend float64) error {
```

Comment:
addend. I'd name it n, d or may be even p. If the value of variable is simply used to add to something you can just name it with simple letters